framework/src/wifi_manager/wifi_manager_api.c

```diff
@@ -100,24 +109,25 @@ wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
 		return WIFI_MANAGER_INVALID_ARGS;
 	}
 
-	wifimgr_msg_s msg = {EVT_INIT_CMD, (void *)wmcb, NULL};
-	wifi_manager_result_e res = wifimgr_post_message(&msg);
+	wifimgr_msg_s msg = {EVT_INIT_CMD, WIFI_MANAGER_FAIL, (void *)wmcb, NULL};
+	int res = wifimgr_post_message(&msg);
 
-	return res;
+	RETURN_RESULT(res, msg);
 }
 
 
 wifi_manager_result_e wifi_manager_deinit(void)
 {
 	sem_t signal;
 	sem_init(&signal, 0, 0);
-	wifimgr_msg_s msg = {EVT_DEINIT_CMD, NULL, &signal};
+	wifimgr_msg_s msg = {EVT_DEINIT_CMD, WIFI_MANAGER_FAIL, NULL, &signal};
 
-	wifi_manager_result_e res = wifimgr_post_message(&msg);
+	int res = wifimgr_post_message(&msg);
```

Comment:
If 'res' is less than zero, do you need to still wait 'msg.signal'?

Reply:
if 'res' is less than zero, then it won't wait 'msg.signal'

Reply:
no it doesn't need to wait signal. I'll fix it